### PR TITLE
Pop-ups, Xs and flips that don't flip

### PR DESCRIPTION
**Pull request by Shea-Scarlet**

### CODE REVIEW

File: frontend/components/QuestionPages/QuestionAnswers.tsx [hunk 1]
```diff
@@ -1,48 +1,115 @@
+import { useState } from 'react';
 import MatchMediaWrapper from '@/components/MatchMediaWrapper';
+import Image from 'next/image';
+import X from 'public/remove 1.png';
```
Comment (jasminetai) [inline frontend/components/QuestionPages/QuestionAnswers.tsx]: Nitpicky, but maybe standardize the asset names somehow -- I see a mix of spaces, hyphens, and uppercase in these imports.

File: frontend/pages/[gameId]/questions/player.tsx [hunk 2]
```diff
@@ -1,27 +1,86 @@
...
+
+      {/* Render the player question bottom */}
+      <PlayerQuestionBottom name={'Student Name'} score={5100}></PlayerQuestionBottom>
+
+      {/* Render the modal when showModal is true */}
+      {showModal && (
+        <div className="fixed top-0 left-0 w-screen h-screen flex items-center justify-center bg-black bg-opacity-50">
+          {/* Render the selected image if selectedAnswer is not null */}
+          {selectedAnswer !== null && (
+            <Image
```
Comment (jasminetai) [inline frontend/pages/[gameId]/questions/player.tsx]: This popup is kinda big at the moment. You could look into the width/height properties and/or setting the width/height using the style attribute for this.

**Review by jasminetai — COMMENTED** [review]
It's still in progress, but looks fine at a glance. I commented a couple small suggestions. Although I see that you are starting the flip animations, which would be cool to see, you might want to focus on displaying the X/check mark appropriately for the right and wrong answers first, as they currently show as all right or all wrong -- functionality's probably the priority.

**Review by jasminetai — DISMISSED** [review]

File: frontend/components/QuestionPages/QuestionAnswers.tsx [hunk 3]
```diff
@@ -1,48 +1,152 @@
...
+            className="absolute bottom-0 left-0 w-588px h-59px flex justify-center items-center"
+            style={{
+              width: '588px',
+              height: '59px',
+              background: '#FF7200',
+              boxShadow: '0px 4px 4px rgba(0, 0, 0, 0.25)',
+              borderRadius: '15px',
+            }}
+            >
+              <p style={{ color: 'white', fontSize: '18px' }}>You Answered</p>
```
Comment (jasminetai) [inline frontend/components/QuestionPages/QuestionAnswers.tsx]: For longer answer texts, this banner should probably go over the answer, not under it. The banner is also too large for smaller screens.

**Review by jasminetai — DISMISSED** [review]
See previous comment on popup size and new comment on "You Answered" banner for potential improvements, but this looks functionally fine.

File: frontend/pages/[gameId]/questions/host.tsx [hunk 4]
```diff
@@ -3,22 +3,28 @@ import QuestionAnswers from '@/components/QuestionPages/QuestionAnswers';
...
+          'To have something to put on my GitHub',
+          'To seek the holy grail',
+        ]}
+        explanations={[
+          'To successfully complete the course 115a',
+          'To develop a functional and professional application',
+          'To showcase my projects on the popular platform GitHub',
+          'To embark on a legendary and noble quest for the holy grail',
+        ]}
+      ></QuestionAnswers>
```
Comment (jasminetai) [inline frontend/pages/[gameId]/questions/host.tsx]: I see that currently, the host version just displays the answer explanation for a clicked answer. Is that intended?

**Review by jasminetai — DISMISSED** [review]
Third time... but this looks good! The answer explanation addition is really nice.

**Review by celdond — DISMISSED** [review]
Looks great, works cleanly.
Nothing really to add.

**Review by celdond — DISMISSED** [review]

**Review by jasminetai — DISMISSED** [review]

**Review by 190n — DISMISSED** [review]

File: package.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
+  "dependencies": {
+    "react-spring": "^9.7.1"
```
Comment (190n) [inline package.json]: shouldn't this be in frontend/package.json?

**Review by celdond — APPROVED** [review]

**Review by jasminetai — APPROVED** [review]

**Pull request merged**